include/openthread/platform/settings.h

```diff
@@ -75,6 +75,10 @@ enum
     OT_SETTINGS_KEY_SRP_SERVER_INFO      = 0x000d, ///< The SRP server info (UDP port).
     OT_SETTINGS_KEY_LEGACY_NAT64_PREFIX  = 0x000e, ///< Reserved. Legacy NAT64 prefix.
     OT_SETTINGS_KEY_BR_ULA_PREFIX        = 0x000f, ///< BR ULA prefix.
+
+    // Keys in range 0x1000-0x7fff are reserved for vendor-specific use.
+    OT_SETTINGS_KEY_VENDOR_RESERVED_BEGIN = 0x1000,
```

Comment:
Two questions:
- Why do vendors need a vendor-specific range?
- Is the range too big?

Reply:
> Why do vendors need a vendor-specific range?

Vendors might want to store some persistent data in vendor-specific extensions. Extensions can be added here: https://github.com/openthread/openthread/blob/3350f598f6759fd7de0c80845a925197458d0420/src/core/CMakeLists.txt#L273

> Is the range too big?

Probably. I think `0x1000-0x1fff` is OK too.

Reply:
The key is defined as `uint16_t aKey` in `otPlatSettings` API. So how about having the key values toward the end of range for vendor use (e.g, the one where last bit is set `0x7fff-0xffff`) instead of the middle range `01000-0x1fff`?

Reply:
`0x8000-0xffff` SGTM. Updated.